src/libs/antares/study/parameters/adq-patch-params.cpp

```diff
@@ -0,0 +1,160 @@
+{
+    CString<24, false> s = PTO_as_string;
+    s.trim();
+    s.toLower();
+    if (s == "dens")
+    {
+        PTO_as_enum = AdequacyPatch::AdqPatchPTO::isDens;
+        return true;
+    }
+    if (s == "load")
+    {
+        PTO_as_enum = AdequacyPatch::AdqPatchPTO::isLoad;
+        return true;
+    }
+
+    logs.warning() << "parameters: invalid price taking order. Got '" << PTO_as_string << "'";
+
+    return false;
+}
+
+bool CurtailmentSharing::updateFromKeyValue(const String& key, const String& value)
+{
+    // Price taking order
+    if (key == "price-taking-order")
+        return StringToPriceTakingOrder(value, priceTakingOrder);
+    // Include Hurdle Cost
+    if (key == "include-hurdle-cost-csr")
+        return value.to<bool>(includeHurdleCost);
+    // Check CSR cost function prior and after CSR
+    if (key == "check-csr-cost-function")
+        return value.to<bool>(checkCsrCostFunction);
+    // Thresholds
+    if (key == "threshold-initiate-curtailment-sharing-rule")
+        return value.to<double>(thresholdRun);
+    if (key == "threshold-display-local-matching-rule-violations")
+        return value.to<double>(thresholdDisplayViolations);
+    if (key == "threshold-csr-variable-bounds-relaxation")
+        return value.to<int>(thresholdVarBoundsRelaxation);
+
+    return false;
+}
+
+const char* PriceTakingOrderToString(AdequacyPatch::AdqPatchPTO pto)
+{
+    switch (pto)
+    {
+    case AdequacyPatch::AdqPatchPTO::isDens:
+        return "DENS";
+    case AdequacyPatch::AdqPatchPTO::isLoad:
+        return "Load";
+    default:
+        return "";
+    }
+}
+
+void CurtailmentSharing::addProperties(IniFile::Section* section) const
+{
+    section->add("price-taking-order", PriceTakingOrderToString(priceTakingOrder));
+    section->add("include-hurdle-cost-csr", includeHurdleCost);
+    section->add("check-csr-cost-function", checkCsrCostFunction);
+
+    // Thresholds
+    section->add("threshold-initiate-curtailment-sharing-rule", thresholdRun);
+    section->add("threshold-display-local-matching-rule-violations", thresholdDisplayViolations);
+    section->add("threshold-csr-variable-bounds-relaxation", thresholdVarBoundsRelaxation);
+}
+
+// ------------------------
+// Adq patch parameters
+// ------------------------
+void AdqPatchParams::reset()
+{
+    enabled = false;
+
+    localMatching.reset();
+    curtailmentSharing.reset();
+}
+
+void AdqPatchParams::addExcludedVariables(std::vector<std::string>& out) const
+{
+    if (!enabled)
+    {
+        out.emplace_back("DENS");
+        out.emplace_back("LMR VIOL.");
+        out.emplace_back("SPIL. ENRG. CSR");
+        out.emplace_back("DTG MRG CSR");
+    }
+}
+
+
+bool AdqPatchParams::updateFromKeyValue(const String& key, const String& value)
+{
+    if (key == "include-adq-patch")
+        return value.to<bool>(enabled);
+
+    return curtailmentSharing.updateFromKeyValue(key, value) || localMatching.updateFromKeyValue(key, value);
```

Comment:
```suggestion
    return curtailmentSharing.updateFromKeyValue(key, value) != localMatching.updateFromKeyValue(key, value); // XOR
```

Reply:
Sorry, I don't see the benefit of **XOR** operator compared to **||** operator

Reply:
- If we leave operator `||`, it means that we accept the key to be added to both `localMatching` and `curtailmentSharing`.
- If we instead use `!=`, it means that the key must be added only to one of these, exclusively

In my opinion, the 2nd option is closer to what we want.

Reply:
This is subtle !
I pushed your suggestion.